p2p/simulations/noderunner.go

```diff
@@ -0,0 +1,250 @@
+// Copyright 2017 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+// Package simulations simulates p2p networks.
+// A NodeRunner simulates starting and stopping real nodes in a network.
+package simulations
+
+import (
+	"fmt"
+	"math/rand"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/p2p/discover"
+	"github.com/julienschmidt/httprouter"
+)
+
+//A NodeRunner only has a Run function
+type NodeRunner interface {
```

Comment:
Did you mean to include the changes from #121 in here too?

Reply:
They are conceptually separate things, that's why I issued separate PRs (which is also simpler to review), but ultimately they both need to be merged - is this the question you are asking?

Reply:
What I mean is, you currently have two PRs and both of them contain the addition of `NodeRunner`, so either there should just be one PR for all of these changes, or two PRs with distinct changes in them.

Reply:
ooooohh I don't think this is what I wanted to do - not sure how noderunner sneaked in here. The idea was that they would be separate (thus I am going to remove noderunner from here)

Reply:
hmmm doesn't look so simple, as removing this file breaks the overlay sim again. Maybe easiest is to provide just one PR, if you don't mind

Reply:
OK, closed #121 as this PR contains all changes from there